includes/Gateway/API/Responses/Create_Payment.php

```diff
@@ -28,7 +28,14 @@ public function transaction_held() {
 
 		// ensure the tender is CAPTURED
 		if ( $this->get_payment() ) {
-			$held = 'AUTHORIZED' === $this->get_payment()->getCardDetails()->getStatus();
+			// Check if the card or wallet is AUTHORIZED (WALLET is for Cash App payments).
+			$card_details   = $this->get_payment()->getCardDetails();
+			$wallet_details = $this->get_payment()->getWalletDetails();
+			if ( ! empty( $card_details ) ) {
+				$held = 'AUTHORIZED' === $card_details->getStatus();
```

Comment:
I think it would be nice to move these values to new constants in the class.

Reply:
Thanks. This is addressed in https://github.com/woocommerce/woocommerce-square/pull/62/commits/c541769d8b9dc0ab88bae34e2cc1b6cb79f7b7db